### PR TITLE
fix: make bupt portal work again

### PR DESCRIPTION
**Pull request by wzekin**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/bupt/portal
/bupt/news
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [ ] 新的路由 New Route
  - [ ] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note

原来的链接失效了， 更换了新的源（需要抓微信登录之后的Cookie

### CODE REVIEW

**Review by TonyRL — COMMENTED** [review]
Please update `university.md` as well and add `selfhost="1"`

File: lib/routes/universities/bupt/news.js [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
-            out.push({
-                title: item.title,
-                description: item.desc,
-                pubDate: new Date(item.created * 1000).toUTCString(),
-                link: `https://webapp.bupt.edu.cn/extensions/wap/news/detail.html?id=${item.id}&classify_id=xnxw`,
-                author: item.author,
-            });
+    const data = response.data;
+    data.forEach((item) => {
+        out.push({
```
Comment (TonyRL) [inline lib/routes/universities/bupt/news.js:21]: You can use
```js
const out = data.map({ title: ..., ...})
```
instead

Comment (wzekin) [reply lib/routes/universities/bupt/news.js]: done~

File: lib/routes/universities/bupt/news.js [hunk 2]
```diff
@@ -1,34 +1,32 @@
 const config = require('@/config').value;
 const got = require('@/utils/got');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const { CookieJar } = require('tough-cookie');
+
+const cookieJar = new CookieJar();
+if (config.bupt.portal_cookie) {
+    cookieJar.setCookie(config.bupt.portal_cookie, 'https://wx.bupt.edu.cn');
+}
```
Comment (TonyRL) [inline lib/routes/universities/bupt/news.js:10]:
```suggestion
if (!config.bupt || !config.bupt.portal_cookie) {
    throw 'BUPT RSS is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
}
cookieJar.setCookie(config.bupt.portal_cookie, 'https://wx.bupt.edu.cn');
```

File: lib/routes/universities/bupt/news.js [hunk 3]
```diff
@@ -1,34 +1,32 @@
...
-        value.forEach((item) => {
-            out.push({
-                title: item.title,
-                description: item.desc,
-                pubDate: new Date(item.created * 1000).toUTCString(),
-                link: `https://webapp.bupt.edu.cn/extensions/wap/news/detail.html?id=${item.id}&classify_id=xnxw`,
-                author: item.author,
-            });
-        });
+        url: `https://wx.bupt.edu.cn/portal-article-page/article-list/news?limit=10&offset=0`,
```
Comment (TonyRL) [inline lib/routes/universities/bupt/news.js:15]:
```suggestion
        url: 'https://wx.bupt.edu.cn/portal-article-page/article-list/news?limit=10&offset=0',
```

File: lib/routes/universities/bupt/portal.js [hunk 4]
```diff
@@ -1,34 +1,32 @@
 const config = require('@/config').value;
 const got = require('@/utils/got');
+const timezone = require('@/utils/timezone');
+const { parseDate } = require('@/utils/parse-date');
+const { CookieJar } = require('tough-cookie');
+
+const cookieJar = new CookieJar();
+if (config.bupt.portal_cookie) {
+    cookieJar.setCookie(config.bupt.portal_cookie, 'https://wx.bupt.edu.cn');
+}
```
Comment (TonyRL) [inline lib/routes/universities/bupt/portal.js:10]:
```suggestion
if (!config.bupt || !config.bupt.portal_cookie) {
    throw 'BUPT RSS is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>';
}
cookieJar.setCookie(config.bupt.portal_cookie, 'https://wx.bupt.edu.cn');
```

File: lib/routes/universities/bupt/portal.js [hunk 5]
```diff
@@ -1,34 +1,32 @@
...
-        value.forEach((item) => {
-            out.push({
-                title: item.title,
-                description: item.text,
-                pubDate: new Date(item.created * 1000).toUTCString(),
-                link: `https://webapp.bupt.edu.cn/extensions/wap/news/detail.html?id=${item.id}&classify_id=tzgg`,
-                author: item.author,
-            });
-        });
+        url: `https://wx.bupt.edu.cn/portal-article-page/article-list/notice?limit=10&offset=0`,
```
Comment (TonyRL) [inline lib/routes/universities/bupt/portal.js:15]:
```suggestion
        url: 'https://wx.bupt.edu.cn/portal-article-page/article-list/notice?limit=10&offset=0',
```

**Review by TonyRL — COMMENTED** [review]
Please provide a screenshot of the RSS feed output

**Comment by DIYgod** [comment]
This PR has been open for too long and will be closed first. If you have any questions, feel free to continue the discussion below.

**Pull request closed without merging**